pkg/apis/build/v1alpha1/clutser_build_template_validation_test.go

```diff
@@ -0,0 +1,111 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestValidateClusterBuildTemplate(t *testing.T) {
+	hasDefault := "has-default"
+	for _, c := range []struct {
+		desc   string
+		tmpl   *ClusterBuildTemplate
+		reason string // if "", expect success.
+	}{{
+		desc: "Single named step",
+		tmpl: &ClusterBuildTemplate{
```

Comment:
I think we can condense both of these test tables into one, and avoid duplication:

```
func TestValidateBuildTemplate(t *testing.T) {
  for _, c := range []struct {
    desc string
    spec BuildTemplateSpec
    reason string
  } {{
   ...
  }} {
    t.Run("namespaced-"+c.desc, func(t *testing.T) {
      tmpl := &BuildTemplate{Spec: c.spec},
      tmpl.Validate()
      ...
    })
    t.Run("cluster-namespaced-"+c.desc, func(t *testing.T) {
      ctmpl := &ClusterBuildTemplate{Spec: c.spec},
      ctmpl.Validate()
      ...
    })
  }
}
```